docs/DeveloperGuide.adoc

```diff
@@ -12,255 +12,359 @@ ifdef::env-github[]
+For example, the `Logic` component defines it's API in the `Logic.java` interface and exposes its functionality using the `LogicManager.java` class.
 
-.Class Diagram of the Logic Component
-image::LogicClassDiagram.png[]
-
-[discrete]
 ==== How the architecture components interact with each other
 
-The _Sequence Diagram_ below shows how the components interact with each other for the scenario where the user issues the command `delete 1`.
+The _Sequence Diagram_ below shows how the components interact with each other for the scenario where the user issues the command `new -t Notably -b Lorem ipsum`.
 
-.Component interactions for `delete 1` command
+.Component interactions for the `new -t Notably -b Lorem ipsum` command
 image::ArchitectureSequenceDiagram.png[]
 
-The sections below give more details of each component.
-
-[[Design-Ui]]
-=== UI component
+[[Design-View]]
+=== View component
 
-.Structure of the UI Component
+.Structure of the View Component
 image::UiClassDiagram.png[]
 
-*API* : link:{repoURL}/src/main/java/seedu/address/ui/Ui.java[`Ui.java`]
+*API* : link:{repoURL}/src/main/java/seedu/address/view/View.java[`View.java`]
 
-The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `PersonListPanel`, `StatusBarFooter` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class.
+The View consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `SuggestionsList`, `SideBarTree`, `HelpModal`, `StatusBarFooter`, `BlockContent` etc. All these, including the `MainWindow`, inherit from the abstract `ViewPart` class.
 
-The `UI` component uses JavaFx UI framework. The layout of these UI parts are defined in matching `.fxml` files that are in the `src/main/resources/view` folder. For example, the layout of the link:{repoURL}/src/main/java/seedu/address/ui/MainWindow.java[`MainWindow`] is specified in link:{repoURL}/src/main/resources/view/MainWindow.fxml[`MainWindow.fxml`]
+The `View` component uses JavaFx framework. The layout of these View parts are defined in matching `.fxml` files that are in the `src/main/resources/view` folder. For example, the layout of the link:{repoURL}/src/main/java/seedu/address/view/MainWindow.java[`MainWindow`] is specified in link:{repoURL}/src/main/resources/view/MainWindow.fxml[`MainWindow.fxml`]
 
-The `UI` component,
+The `View` component,
 
 * Executes user commands using the `Logic` component.
-* Listens for changes to `Model` data so that the UI can be updated with the modified data.
+* Listens for changes to `Model` data so that the View can be updated with the modified data.
 
 [[Design-Logic]]
 === Logic component
 
-[[fig-LogicClassDiagram]]
-.Structure of the Logic Component
-image::LogicClassDiagram.png[]
+.Architecture of Logic
+image::LogicArchitectureDiagram.png[]
```

Comment:
Cool diagram! However, some features of a class diagram are present (dotted arrows, aggregation). Since it is a architecture diagram, it is not clear what they mean. Again, aim to capture high-level details in architecture diagrams. Unless you meant to make it a class diagram, where you would need to add more details and clarity. Maybe referring to AB3 architecture diagrams will be helpful.

Reply:
Agreed with the previous comment! There are many interfaces on the architecture diagram, perhaps reducing some would be a start. However, overall the logic was fine and everything made sense. One suggestion from me would be perhaps using straight arrows instead of curved ones as it may make your diagram neater.